src/App.jsx

```diff
@@ -14,33 +17,91 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button type="button" className="button is-info is-light">
-        Sort alphabetically
-      </button>
+const SORTING_TYPE_NAME = 'alphabetical';
+const SORTING_TYPE_LENGTH = 'length';
 
-      <button type="button" className="button is-success is-light">
-        Sort by length
-      </button>
+function getPreparedGoods(goods, sortingType) {
+  const preparedGoods = [...goods];
 
-      <button type="button" className="button is-warning is-light">
-        Reverse
-      </button>
+  if (sortingType) {
+    preparedGoods.sort((good1, good2) => {
+      switch (sortingType) {
+        case SORTING_TYPE_NAME:
+          return good1.localeCompare(good2);
+        case SORTING_TYPE_LENGTH:
+          return good1.length - good2.length;
+        default:
+          return 0;
+      }
+    });
+  }
 
-      <button type="button" className="button is-danger is-light">
-        Reset
-      </button>
-    </div>
+  return preparedGoods;
+}
+
+export const App = () => {
+  const [sortingType, setSortingType] = useState('');
+  let visibleGoods = getPreparedGoods(goodsFromServer, sortingType);
+  const [reversedState, setReversedState] = useState(false);
+
+  if (reversedState) {
+    visibleGoods = visibleGoods.reverse();
+  }
+
+  return (
+    <div className="section content">
+      <div className="buttons">
+        <button
+          type="button"
+          className={classNames('button', 'is-info', {
+            'is-light': sortingType !== SORTING_TYPE_NAME,
+          })}
+          onClick={() => setSortingType(SORTING_TYPE_NAME)}
+        >
+          Sort alphabetically
+        </button>
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+        <button
+          type="button"
+          className={classNames('button', 'is-success', {
+            'is-light': sortingType !== SORTING_TYPE_LENGTH,
+          })}
+          onClick={() => setSortingType(SORTING_TYPE_LENGTH)}
+        >
+          Sort by length
+        </button>
+
+        <button
+          type="button"
+          className={classNames('button', 'is-warning', {
+            'is-light': !reversedState,
+          })}
+          onClick={() => setReversedState(!reversedState)}
+        >
+          Reverse
+        </button>
+
+        {(sortingType || reversedState) && (
+          <button
+            type="button"
+            className="button is-danger is-light"
+            onClick={() => {
+              setSortingType('');
+              setReversedState(false);
+            }}
```

Comment:
Move this logic from jsx to the helper reset function and use it here
```suggestion
            onClick={reset}
```